Sources/MapboxNavigation/NavigationView.swift

```diff
@@ -66,20 +66,18 @@ open class NavigationView: UIView {
         navigationMapView.delegate = delegate
     }
     
-    var tileStoreLocation: TileStoreConfiguration.Location? = .default
-    private var _navigationMapView: NavigationMapView? = nil
-    lazy var navigationMapView: NavigationMapView = {
-        _navigationMapView?.frame = bounds
-        let navigationMapView = _navigationMapView ?? NavigationMapView(frame: bounds, tileStoreLocation: tileStoreLocation)
-        navigationMapView.isHidden = false
-        navigationMapView.translatesAutoresizingMaskIntoConstraints = false
-        
-        navigationMapView.delegate = delegate
-        navigationMapView.navigationCamera.viewportDataSource = NavigationViewportDataSource(navigationMapView.mapView,
-                                                                                             viewportDataSourceType: .active)
-        
-        return navigationMapView
-    }()
+    // :nodoc:
+    public var navigationMapView: NavigationMapView {
```

Comment:
Unfortunately I do not see any other ways of implementing seamless transition other than replacing `navigationMapView` instance on the fly. Original approach that allows to pass `NavigationMapView` instance via `NavigationOptions.navigationMapView` property creates "blinking" effect that is not acceptable in my opinion.

Reply:
Could we address the blinking by having `NavigationOptions.navigationMapView` modify this property in the animator instead of injecting the map view when it currently does? I guess that would require encapsulating more of the animation inside the navigation SDK as opposed to application code.

Reply:
I don't think that blink will disappear when using `NavigationOptions.navigationMapView`, here's what happens when we use such approach with custom transition:

https://user-images.githubusercontent.com/1496498/181088595-4361f0ff-6ed8-40f9-b57c-c08209e9d72f.mov

As you can see there is small period of time when view object is transferred via `NavigationOptions` to `NavigationViewController` and is already removed from view hierarchy. This happens due to object transfer in `NavigationView` initializer.

Here's what happens when we access `NavigationMapView` object directly from presenter and do the transfer in the `PresentationAnimator` and do not pass `NavigationMapView` to `NavigationOptions`:

https://user-images.githubusercontent.com/1496498/181090890-631c3f8c-254c-48da-9c76-b1f8c7fa2ca9.mov

If we do not use `NavigationOptions`, but want to encapsulate more animation logic into SDK we'd need to completely wrap `NavigationViewController` presentation in Navigation SDK APIs, reason for this is that there are several requirements for transition to work:
1. `NavigationViewController` presentation should be always animated (i.e. `present(navigationViewController, animated: true) should be called`).
2. `NavigationViewController.transitioningDelegate` property should be set.
3. `NavigationViewController.modalPresentationStyle` should be preferably set to `.fullScreen`.

This will mean that `DismissalAnimator` and `PresentationAnimator` will reside in SDK as well. If we follow such approach it'll mean that `NavigationViewController` presentation will not be as agile as a current one, but easier to understand (we can try to expose something like `NavigationViewController.present()`).

Reply:
OK, thanks for documenting what you found with these alternative approaches. As tail work, maybe we should deprecate `NavigationOptions.navigationMapView` in favor of this option, once it’s documented, so that developers can find it more easily.